extensions/shex-map/lib/extensions.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * This file is the main entry point into calling an extension. 
+ * It determines which extension is requested, and then, assuming
+ * the extension is valid, it forwards the request on
+ */
+
+var _ = require('underscore');
+
+// Known extensions
+var regex_extension = require('./regex_extension');
+
+var utils = require('./extension-utils');
+
+/**
+ * Given a map directive that contains an extension of format 
+ *          extensionName(args) 
+ * split it up for easy access to extenion name and arguments separately 
+ * 
+ * @param a map directive with an extension call embedded
+ *
+ * @return an object with members:  extension name and the arguments.
+ */
+function extensionDef(mapDirective) { 
+    if (_.isEmpty(mapDirective)) 
+        throw Error("Invalid extension function: " + mapDirective + "!");
+
+    // Get the extension name and argument(s)  
+    mapDirective = mapDirective.trim(); // Strip any leading or trailing white space
+    var startArgs = mapDirective.indexOf('(', 0);
+    var endArgs = mapDirective.lastIndexOf(')');
+    if (startArgs < 2 || endArgs < 4 || endArgs <= startArgs+1 || endArgs != mapDirective.length-1) 
+        throw Error("Invalid extension function: " + mapDirective + "!");
+
+    return { name:  mapDirective.substring(0, startArgs),
+             args:  mapDirective.substring(startArgs+1, endArgs) }; 
+}
+
+function lift(mapDirective, input, prefixes) {
```

Comment:
I don't care what we call it.  In his earlier references that I built this from, he called it lift/lower.  Eric, can you weigh in on this?